.github/workflows/release.yaml

```diff
@@ -83,12 +83,33 @@ jobs:
       registry-password: ${{ secrets.GITHUB_TOKEN }}
 
   validate:
-    needs: provenance
+    needs: [provenance, build]
     runs-on: ubuntu-latest
     steps:
       - name: Validate image
         uses: enterprise-contract/action-validate-image@latest
         with:
-          image: ${{ env.IMAGE_REGISTRY }}/${{ env.IMAGE_REPO }}:${{ env.DIGEST }}
+          image: ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
           identity: https:\/\/github\.com\/(slsa-framework\/slsa-github-generator|${{ github.repository_owner }}\/${{ github.event.repository.name }})\/
           issuer: https://token.actions.githubusercontent.com
+    
+  Add-Tag-Validate:
+    runs-on: ubuntu-latest
+    needs: [validate, build]
+    steps:  
+      # needed to login into ghcr.io and push the tag
+      - name: Log in to ghcr
+        uses: redhat-actions/podman-login@9184318aae1ee5034fbfbacc0388acf12669171f # v1
+        with:
+          registry: ${{ env.IMAGE_REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Pull image using Buildah
+        run: buildah pull ${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }}
```

Comment:
The `pull-tag-push` approach to add a tag is old school :stuck_out_tongue_closed_eyes: 

It would be great if this used skopeo instead:

```bash
skopeo copy \
  docker://${{ needs.build.outputs.image }}@${{ needs.build.outputs.digest }} \
  docker://${{ needs.build.outputs.image }}:latest
```

The above tags the image with the `latest` tag.

Skopeo should already be installed on the [runner image](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md).